cmd/exec/operations.go

```diff
@@ -497,22 +516,26 @@ func executeEvent(url string) (string, error) {
 		)
 
 		if err != nil {
-			return event["id"].(string), errors.New("filtetype not json")
+			return "", errors.New("direktiv server not reachable")
 		}
 
 		addAuthHeaders(req)
 
 		resp, err := http.DefaultClient.Do(req)
 		if err != nil {
-			return event["id"].(string), err
+			return "", err
 		}
 
 		if resp.StatusCode != http.StatusOK {
-			return event["id"].(string), errors.New("Bad request")
+			return "", errors.New("Bad request")
 		}
 	}
 
-	return event["id"].(string), err
+	if event["id"] != nil {
+		return event["id"].(string), nil
```

Comment:
This line can still crash. It's good that you're checking that `id` exists. But what if it's like this?

```
{
  "id": [1, 2, 3]
}
```

It exists, but it won't successfully assert as a string.

We can check if a type assertion succeeds similar to how you can test if a value exists in a map:

```
s, ok := x.(string)
if !ok {
  // it wasn't a string
}
```

But in this case the better option might be to use the offical Go cloudevents library to perform this validation when it loads the file.